content-learn/mastering-meshery-istio/advance-concepts-of-service-mesh/istio/deploy-an-application.mdx

```diff
@@ -0,0 +1,218 @@
+<a href={BookInfoOnMesh}>
+  <img src={BookInfoOnMesh} width="100%" align="center" />
+</a>
+
+_Figure: BookInfo deployed on the mesh_
+
+<br />
+Sidecars proxy can be either manually or automatically injected into the pods. Automatic
+sidecar injection requires that your Kubernetes api-server supports `admissionregistration.k8s.io/v1`
+or `admissionregistration.k8s.io/v1beta1` or `admissionregistration.k8s.io/v1beta2`
+APIs. Verify whether your Kubernetes deployment supports these APIs by executing:
+
+```sh
+kubectl api-versions | grep admissionregistration
+```
+
+If your environment **does NOT** supports either of these two APIs, then you may use [manual sidecar injection](#manual-sidecar-inj) to deploy the sample app.
+
+As part of Istio deployment in [Previous chapter](/learn-ng/mastering-meshery-istio/introduction-to-service-meshes/istio/getting-started), you have deployed the sidecar injector.
+
+<br />
+<h2 class="chapter-sub-heading">
+  <a name="auto"></a> Deploying Sample App with Automatic sidecar injection
+</h2>
+
+<br />
+Istio, deployed as part of this workshop, will also deploy the sidecar injector.
+Let us now verify sidecar injector deployment.
+
+```sh
+kubectl -n istio-system get configmaps istio-sidecar-injector
+```
+
+Output:
+
+```sh
+NAME                     DATA   AGE
+istio-sidecar-injector   2      9h
+```
+
+NamespaceSelector decides whether to run the webhook on an object based on whether the namespace for that object matches the [selector](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors).
+
+```sh
+kubectl get namespace -L istio-injection
+```
+
+Output:
+
+```sh
+NAME           STATUS    AGE       ISTIO-INJECTION
+default        Active    1h        enabled
+istio-system   Active    1h        disabled
+kube-public    Active    1h
+kube-system    Active    1h
+```
+
+Using Meshery, navigate to the Istio management page.
+
+1. Enter `default` in the `Namespace` field.
+1. Click the (+) icon on the `Sample Application` card and select `BookInfo Application` from the list.
+
+This will do 3 things:
+
+1. Label `default` namespace for sidecar injection.
+1. Deploys all the BookInfo services in the `default` namespace.
+1. Deploys the virtual service and gateway needed to expose the BookInfo's productpage application in the `default` namespace.
+
+<h3 class="chapter-sub-heading">
+  <a name="verify"></a> Verify Bookinfo deployment{" "}
+</h3>
+
+1. Verify that the deployments are all in a state of AVAILABLE before continuing.
+
+   ```sh
+   watch kubectl get deployment
+   ```
+
+2. Choose a service, for instance `productpage`, and view it's container configuration:
+
+```sh
+kubectl get po
+
+kubectl describe pod productpage-v1-.....
+```
+
+3. Examine details of the services:
+
+   ```sh
+   kubectl describe svc productpage
+   ```
+
+Next, you will expose the BookInfo application to be accessed external from the cluster.
+
+<br />
+<h3>Alternative: Manual installation </h3>
```

Comment:
Move this `Alternative` section below the Automatic section, that is after line:108

Reply:
Also, please make this update to chapter-2 of the `introduction-to-service-mesh` course as well.

Reply:
Sorry for the confusion, but please move it back to where it was earlier. It doesn't go with the context above.

Reply:
Hey @Nikhil-Ladha ,please do tell me if there is conflict or not , most of the time I'am having difficulty with git only.